### ISSUE TITLE
Marketing->Sitemap Grid Not Showing and Sitemap generation not running

### ISSUE THREAD

**Issue by devonto** [magento/magento2#7586]
I noticed that the sitemap.xml file which was previously configured was not being regenerated.

When checking the settings in "Marketing"->"Sitemap", a javascript error appears preventing the page loading fully.

### Preconditions
1. Magento v2.1.1
2. A sitemap was previously created on an older version
3. Javascript bundling is on
4. Site is in production mode

### Steps to reproduce
1. Go to "Marketing" then "Sitemap"

### Expected result
1. List of current sitemaps should be displayed

### Actual result
1. Javascript error displayed and only basics of page is loaded
2. Note the "1 records found", verifying there is a sitemap to load

```
VM8674:18 Uncaught TypeError: Cannot read property 'select' of undefined(…)
bindFieldsChange @ VM8674:18
initGrid @ VM8674:6
initialize @ VM8674:2
g @ VM8678:2
(anonymous function) @ (index):375
execCb @ 5354340….min.js:112
check @ 5354340….min.js:56
enable @ 5354340….min.js:73
init @ 5354340….min.js:54
(anonymous function) @ 5354340….min.js:97
```
![site-map-_-seo- -search-_-marketing](https://cloud.githubusercontent.com/assets/6807236/20664680/1f563b26-b554-11e6-8d11-e24a7389c354.png)

**Comment by shiftedreality** [comment]
Hi @devonto 

I was unable to reproduce your issue with upgrading from 2.0.9 to 2.1.1
Please, clean your Magento and browser caches.
If the problem still exists-provide more details about your environment configuration.

**Comment by devonto** [comment]
Hi @shiftedreality 

All caches are clean and all content recompiled fresh - I also turned off all Javascript minification/bundling/merging

This has pinpointed the error to grid.js on line 284
`grid.js:284 Uncaught TypeError: Cannot read property 'select' of undefined(…)`

As I originally mentioned, the sitemap.xml file itself is not being generated - not sure now if this grid not displaying is an artefact of a larger issue.

Is there a way to trigger sitemap generation manually?

**Comment by devonto** [comment]
Additionally, the following row exists in the database table `sitemap`.

```
sitemap_id	sitemap_type	sitemap_filename	sitemap_path	sitemap_time	store_id
12		(NULL)		sitemap.xml		/		(NULL)		1
```

**Comment by devonto** [comment]
I have resolved my issue, which took a good bit of digging.

Within `vendor/magento/module-sitemap/Model/Sitemap.php`, the function `_getDocumentRoot()` was not returning a value as `$this->_request->getServer('DOCUMENT_ROOT')` was not set.

I have changed this to return `realpath(dirname(__FILE__))`.

This may be a specific server configuration at my end, however it would be good to be picked up and looked at for future releases.
